app/authorization.go

```diff
@@ -61,6 +72,12 @@ func (a *App) SessionHasPermissionToChannel(session model.Session, channelId str
 	}
 
 	channel, err := a.GetChannel(channelId)
+	if err != nil {
+		return false
+	}
+	if session.IsUnrestricted() {
+		return true
+	}
```

Comment:
@reflog I moved this check lower so that there's parity between the behaviour of an unrestricted session and a normal role when the channel id does not exist.